galera/src/certification.cpp

```diff
@@ -1075,7 +1076,7 @@ wsrep_seqno_t galera::Certification::set_trx_committed(TrxHandle* trx)
             deps_set_.erase(i);
         }
 
-        if (gu_unlikely(index_purge_required()))
+        if (gu_unlikely(gcache_.cleanup_required() || index_purge_required()))
```

Comment:
Correctness should be the same because it only deletes pages that aren't in use.  The default behavior (when no constraints are set) will have the same performance.

If they do set the parameters, as long as haven't hit the limits, the behavior will be the same.  Once we hit one of the constraints, we'll be discarding the seqno's more often and calling delete_page() more often.  But this is a direct consequence of using an OR condition rather than an AND.